operators.py

```diff
@@ -330,49 +357,59 @@ def execute(self, context):
             tmp_tex_cmds = get_texture_list(rpass.scene)
             rpass.update_frame_num(scene.frame_start)
             tmp2_cmds = get_texture_list(rpass.scene)
-            job_tex_cmds = [cmd for cmd in tmp_tex_cmds if cmd in tmp2_cmds]
+            if rm.convert_textures:
+                job_tex_cmds = [
+                    cmd for cmd in tmp_tex_cmds if cmd in tmp2_cmds]
 
             for frame in range(scene.frame_start, scene.frame_end + 1):
                 rpass.update_frame_num(frame)
-                self.report(
-                    {'INFO'}, 'RenderMan External Rendering generating rib for frame %d' % frame)
-                self.gen_rib_frame(rpass)
+                if do_rib:
+                    self.report(
+                        {'INFO'}, 'RenderMan External Rendering generating rib for frame %d' % scene.frame_current)
+                    self.gen_rib_frame(rpass, do_objects)
                 rib_names.append(rpass.paths['rib_output'])
-                frame_tex_cmds[frame] = [cmd for cmd in get_texture_list(
-                    rpass.scene) if cmd not in job_tex_cmds]
+                if rm.convert_textures:
+                    frame_tex_cmds[frame] = [cmd for cmd in get_texture_list(
+                        rpass.scene) if cmd not in job_tex_cmds]
                 if rm.external_denoise:
                     denoise_files.append(rpass.get_denoise_names())
-                    if rpass.aov_denoise_files:
-                        denoise_aov_files.append(rpass.aov_denoise_files)
+                    if rm.spool_denoise_aov:
+                        denoise_aov_files.append(
+                            self.gen_denoise_aov_name(scene, rpass))
 
         else:
-            self.report(
-                {'INFO'}, 'RenderMan External Rendering generating rib for frame %d' % scene.frame_current)
-            self.gen_rib_frame(rpass)
+            if do_rib:
+                self.report(
+                    {'INFO'}, 'RenderMan External Rendering generating rib for frame %d' % scene.frame_current)
+                self.gen_rib_frame(rpass, do_objects)
             rib_names.append(rpass.paths['rib_output'])
-            frame_tex_cmds = {scene.frame_current: get_texture_list(scene)}
+            if rm.convert_textures:
+                frame_tex_cmds = {scene.frame_current: get_texture_list(scene)}
             if rm.external_denoise:
                 denoise_files.append(rpass.get_denoise_names())
-                if rpass.aov_denoise_files:
-                    denoise_aov_files.append(rpass.aov_denoise_files)
+                if rm.spool_denoise_aov:
+                    denoise_aov_files.append(
```

Comment:
Doesn't have to be in this case.  The aov_denoise_files function returns a single item that is a list of filenames.  Since it only adds that one item to the 'master list' (denoise_aov_files) it works fine.

If you looked at the denoise_aov_files list once its run it would be something like this:
[[Scene.aov.001_diffuse, Scene.aov.001_specular], [Scene.aov.002_diffuse, Scene.aov.002_specular], [Scene.aov.003_diffuse, Scene.aov.003_specular]] and so on.